typing/jkind.ml

```diff
@@ -2086,7 +2094,17 @@ let get_modal_upper_bounds (type l r) ~jkind_of_type (jk : (l * r) jkind) :
   let ( ({ layout = _; mod_bounds; with_bounds = No_with_bounds } :
           Allowance.right_only jkind_desc),
         _ ) =
-    Jkind_desc.normalize ~mode:Ignore_best ~jkind_of_type jk.jkind
+    Jkind_desc.map_normalize ~mode:Ignore_best ~jkind_of_type
+      ~map_type_info:(fun _ ti ->
+        { relevant_axes =
+            (* Optimization: We only care about comonadic modal axes *)
+            Axis_set.intersection ti.relevant_axes
+              (Axis_set.create ~f:(fun ~axis ->
+                   match axis with
+                   | Pack (Modal (Comonadic _)) -> true
+                   | _ -> false))
```

Comment:
(nit) I don't like wildcard matches, but feel free to ignore

Reply:
I also don't like them. Changed.